tests/conftest.py

```diff
@@ -24,6 +25,14 @@
 pytest_plugins = ["pytester"]
 
 
+@pytest.fixture(autouse=True)
+def set_log_level_debug(caplog):
+    # Set the post-test log level to DEBUG for failing tests.  For all tests
+    # (failing and successful), the live log level can be additionally set in
+    # `setup.cfg`. It should be set to WARNING.
+    caplog.set_level(logging.DEBUG)
```

Comment:
for my understanding: even if the log level is e.g. `warning`, pytest will capture it anyway and print it on a failure?

Reply:
(Taken from [here](https://docs.pytest.org/en/latest/logging.html))

According to the docs, pytest only shows the capture log for failing tests. So in this case, we are setting the capture log level to `DEBUG`, and therefore will be able to see all output from failing tests.

The live logs on the other hand are configured to WARNING. This way we can see which tests have succeeded but with warnings.

Reply:
I tested this manually by creating two tests: one that always failed and another one that didn't.  I included calls to `logging.debug` and `logging.warning` on both to see what happened. On the successful test, only the warning was shown (on the live log). For the failing test, the warning was shown on the live log, and after both tests had ran, all of its output (warnings + debug) were shown as well.

Reply:
nice! great solution!